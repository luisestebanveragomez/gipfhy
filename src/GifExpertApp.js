import React from 'react';
import { useState } from 'react';

import { GifGrid } from './components/GifGrid';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = ({ defaultCategories = [] }) => {

  const [categories, setCategories]  = useState(defaultCategories)

  return (
    <>
      <h2>GiftExpertApp</h2>
      <AddCategory setCategories={ setCategories } />
      <hr/>

      <ol>
        {
          categories.map( category  => (
              <GifGrid
                key={ category }
                category={ category }
              />
          ))
        }
      </ol>
    </>
  )
}