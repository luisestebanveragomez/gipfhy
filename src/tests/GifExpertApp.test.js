import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';

import { GifExpertApp } from '../GifExpertApp';

describe('pruebas en <GifExpertApp />', () => {

  test('debe mostrarse correctamente', () => {

    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();

  });

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['halo', 'hola'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe( categories.length )
  })

});