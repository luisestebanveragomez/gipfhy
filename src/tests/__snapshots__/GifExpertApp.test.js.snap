// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="halo"
      key="halo"
    />
    <GifGrid
      category="hola"
      key="hola"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en <GifExpertApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
